src/bin/tree-sitter-graph/main.rs

```diff
@@ -60,6 +61,11 @@ fn main() -> Result<()> {
     let graph = file
         .execute(&ctx, &tree, &source, &mut functions, &mut globals)
         .with_context(|| format!("Could not execute TSG file {}", tsg_path.display()))?;
-    print!("{}", graph.display_with(&ctx));
+    let json = matches.is_present("json");
+    if json {
+        graph.display_json(&ctx);
+    } else {
+        print!("{}", graph.display_with(&ctx));
+    }
```

Comment:
@patrickt wisely suggested that we branch here rather than try to shoehorn JSON formatting into the `display_with` machinery.

Reply:
:100:  Hard agree